.github/workflows/ci.yml

```diff
@@ -3,26 +3,20 @@ name: CI
 on:
   push:
     paths:
-      - '.github/workflows/ci.yaml'
       - 'Mifare Classic Tool/**'
   pull_request:
     paths:
-      - '.github/workflows/ci.yaml'
       - 'Mifare Classic Tool/**'
 
 env:
   PROJECT_PATH: Mifare Classic Tool
 
 jobs:
   build:
-    name: Build
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/cache@v2
-        with:
-          path: ~/.gradle/caches
-          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle/wrapper/gradle-wrapper.properties', '**/buildSrc/src/main/kotlin/**.kt') }}
+      - uses: actions/checkout@v3
+      - uses: gradle/gradle-build-action@v2
```

Comment:
https://github.com/gradle/gradle-build-action

It's really so fast to run jobs and many other advantages.